### PR TITLE
general: Added bulk_rename

### PR DESCRIPTION
**Pull request by dylanaraps**
Closes #71

### CODE REVIEW

File: fff [hunk 1]
```diff
@@ -368,6 +369,50 @@ trash() {
...
+    }
+
+    printf '%s\n%s\n' \
+        "# This file will be executed when the editor is closed." \
+        "# Clear the file to abort." > "$rename_file"
+
+    # Construct the rename commands.
+    for ((i=0;i<${#marked_files[@]};i++)); {
+        [[ ${marked_files[i]} != "${changed_files[i]}" ]] && {
+            printf 'mv -- "%s" "%s"\n' "${marked_files[i]}" "${changed_files[i]}"
```
Comment (ppppisme) [inline fff]: I guess we still need -i flag, so user doesn't overwrite already existing file by accident

File: fff [hunk 2]
```diff
@@ -368,6 +369,50 @@ trash() {
...
+    printf '%s\n' "${marked_files[@]}" > "$rename_file"
+    "${VISUAL:-${EDITOR:-vi}}" "$rename_file"
+
+    # Read the renamed files to an array.
+    IFS=$'\n' read -d "" -ra changed_files < "$rename_file"
+
+    # If the user deleted a line, stop here.
+    ((${#marked_files[@]} != ${#changed_files[@]})) && {
+        rm "$rename_file"
+        cmd_line "warn: Line mismatch in rename file. Doing nothing."
```
Comment (ppppisme) [inline fff]: For me, since it causes the function to abort, it's an error type of message, not warning

File: fff (hunk 1 above)
Comment (dylanaraps) [reply fff]: Check the latest commit. I swapped to `[[ ! -f ${changed_files[i]} ]]` (if the new filename doesn't already exist, add the command to the list).

File: fff (hunk 2 above)
Comment (dylanaraps) [reply fff]: Will do.

**Comment by ppppisme** [comment]
Thanks for your effort, your implementation is much better. I'll be happy to see it in master. Hope I could be helpful

**Comment by ppppisme** [comment]
I've noticed a bug: bulk_rename doesn't work correctly if file has double quote(s) in its name:
```
mv -i -- "/home/pppp/tmp/fff/"README.md"" "/home/pppp/tmp/fff/"README.md"123"
```

**Comment by dylanaraps** [comment]
Hm. I quoted the paths to fix issues with spaces in the names. What I'll do is escape the spaces instead. :+1:

**Comment by dylanaraps** [comment]
Fixed. Also added `-i` to `y` and `m` operations. :)

**Comment by ppppisme** [comment]
Will it work with the other symbols that are also should be escaped? I guess it's better to use printf with '%q' format here

**Comment by dylanaraps** [comment]
Fixed.

**Comment by dylanaraps** [comment]
Thanks :+1:

**Pull request merged**